app/scripts/main.js

```diff
@@ -236,28 +236,69 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
       var self_ = this;
       var isPatientAttributes = (type === 'patient');
       var _data = isPatientAttributes ? patientData_ : sampleData_;
-      var hasAttrDataMap = isPatientAttributes ? hasPatientAttrDataMap_ : hasSampleAttrDataMap_;
+      var hasAttrDataMap = isPatientAttributes ?
+        hasPatientAttrDataMap_ : hasSampleAttrDataMap_;
+
+      $.when(
+        window.iviz.datamanager.getClinicalData(attrIds, isPatientAttributes))
+        .then(function(clinicalData) {
+          var _caseIdToClinDataMap = {};
+          var idType = isPatientAttributes ? 'patient_id' : 'sample_id';
+          _.each(clinicalData, function(_clinicalAttributeData, _attrId) {
+            var attrs = isPatientAttributes ? data_.groups.patient.attr_meta :
+              data_.groups.sample.attr_meta;
+            var selectedAttrMetaIndex = -1;
+
+            _.every(attrs, function(attr, index) {
```

Comment:
You can use charts variable to get attribute object(something like  charts[_attrId]) instead of looping all attributes.